.github/workflows/ci-python.yml

```diff
@@ -48,6 +48,11 @@ jobs:
     # 20.04 is required to install PostgreSQL 12
     runs-on: ubuntu-20.04
     steps:
+      - name: Retrieve runner specs
+        id: runner-specs
+        uses: kenchan0130/actions-system-info@b98fe44bc35efcc60a9828cf4a04783a8b370bde # pin v1.3.0
```

Comment:
Can we avoid using this external action ?
My issue is that the repos is not very popular (only 6 stars) and has a [lot of dependencies](https://github.com/kenchan0130/actions-system-info/blob/master/package-lock.json) so I'm not confident in adding it to our codebase...

My guess is we can achieve something similar by doing;
```
name: Retrieve runner specs
id: runner-specs
run: |
  python -c '
    import platform
    # allowed values are https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context
    match "${runner.os}":
        case "Linux":
            r = platform.freedesktop_os_release()
            # e.g. `ubuntu-22.04`
            os_id = f"{r["ID"]}-{r["VERSION_ID"]}"
        case "macOS":
            # e.g. `macos-13.6.7`
            os_id = "macos-" + platform.mac_ver()[0]
        case "Windows":
            # e.g. `windows-10`
            os_id = "windows-" + platform.win32_ver()[0]
        case unknown:
            raise ValueError(f"Unknown platform {unknown!r}")
    print("::set-output name=os_id::" + os_id)
  '
```